toast_planck/reproc_ring.py

```diff
@@ -1955,6 +2006,20 @@ def build_templates(self, rings):
             self.mapsampler_freq = mapsampler_fg
             self.mapsampler_freq_has_dipole = False
 
+        # Load FSL beam mask and corresponding low resolution sky model (one for each detector)
+        if self.fslbeam_mask_path is not None:
+            if self.fslbeam_mask is None:
+                self.fslbeam_mask = {}
+                self.nside_fsl = {}
+                self.fsl_pixels = {}
+                if self.downgraded_mapsampler_freq is None:
+                    self.downgraded_mapsampler_freq = {}
+                for det in self.dets:
+                    self.fslbeam_mask[det] = hp.read_map(self.fslbeam_mask_path[det])
+                    self.nside_fsl[det] = hp.get_nside(self.fslbeam_mask[det])
+                    self.fsl_pixels[det] = np.arange(12*self.nside_fsl[det]**2)[self.fslbeam_mask[det]]
```

Comment:
You should cast `fslbeam_mask[det]` into a boolean array. Perhaps on line 2018:
```self.fslbeam_mask[det] = hp.read_map(self.fslbeam_mask_path[det]) != 0```. Isn't the array is integer-valued here?